index.js

```diff
@@ -58,51 +91,39 @@
         }
       }
     }
-    const name = getMediaTitle(node)
+
+    const result = {
+      'name': getMediaTitle(node),
+      src,
+      srcIsMediaSourceObjectURL,
+      'pageSrc': window.location.href,
+      'pageTitle': document.title,
+      mimeType,
+      'duration': getMediaDurationInSeconds(node),
+      'detected': true
+    }
 
     if (src) {
-      return [{
-        'name': name,
-        src,
-        'pageSrc': window.location.href,
-        'pageTitle': document.title,
-        'mimeType': mimeType,
-        'duration': getMediaDurationInSeconds(node),
-        'detected': true
-      }]
-    } else {
-      const target = node
-      const sources = []
-      document.querySelectorAll('source').forEach(function (node) {
-        const src = fixUpUrl(node.src)
-        if (src) {
-          if (node.closest('video') === target) {
-            sources.push({
-              'name': name,
-              src,
-              'pageSrc': window.location.href,
-              'pageTitle': document.title,
-              'mimeType': mimeType,
-              'duration': getMediaDurationInSeconds(target),
-              'detected': true
-            })
-          }
-
-          if (node.closest('audio') === target) {
-            sources.push({
-              'name': name,
-              src,
-              'pageSrc': window.location.href,
-              'pageTitle': document.title,
-              'mimeType': mimeType,
-              'duration': getMediaDurationInSeconds(target),
-              'detected': true
-            })
-          }
+      return [result]
+    }
+
+    const target = node
+    const sources = []
+    for (const node of document.querySelectorAll('source')) {
+      const source = { ...result }
+      source.src = fixUpUrl(node.src)
```

Comment:
Hey sorry, I can't comment in `fixUpUrl` but I was wondering if 

`new URL(url, window.location.origin)` should be `new URL(url, window.location)` to get the same behavior as urls on a page?

`foo` on `https://example.com/baz` should go to `https://example.com/baz/foo` but on `https://example.com` it should be `https://example.com/foo`

Reply:
Aha, current implementation ignores the subpath and gives wrong absolute url, doesn't it? Good point! Done.